payloads/remote_shell.py

```diff
@@ -0,0 +1,236 @@
+        out = "OK: %s\n" % msg if ok else "FAIL: %s\n" % msg
+        return out, ""
+
+    if stripped == ".syscalls":
+        names = sorted(SYSCALL.keys())
+        out = "Syscalls definidas:\n" + ", ".join(names) + "\n"
+        return out, ""
+
+    # .vars -> mostrar keys de ctx
+    if stripped == ".vars":
+        names = sorted(ctx.keys())
+        out = "Context variables:\n" + ", ".join(names) + "\n"
+        return out, ""
+
+    # .type nombre
+    if stripped.startswith(".type "):
+        name = stripped[6:].strip()
+        if name in ctx:
+            obj = ctx[name]
+            out = "type(%s): %r\n" % (name, type(obj))
+        else:
+            out = "Name %r not found in context.\n" % name
+        return out, ""
+
+    # .repr nombre
+    if stripped.startswith(".repr "):
+        name = stripped[6:].strip()
+        if name in ctx:
+            obj = ctx[name]
+            out = "repr(%s): %r\n" % (name, obj)
+        else:
+            out = "Name %r not found in context.\n" % name
+        return out, ""
+
+    # Ningún comando especial
+    return None, None
+
+
+def handle_command(cmd, ctx):
+    log("handle_command() received cmd=%r" % cmd)
+
+    cmd = cmd.rstrip("\n")
+    if not cmd:
+        log("Empty command.")
+        return "", ""
+
+    # Primero intentamos comandos especiales tipo .dir, .vars, etc.
+    special_out, special_err = handle_special_commands(cmd, ctx)
+    if special_out is not None or special_err is not None:
+        return special_out or "", special_err or ""
+
+    stdout_buf = StringIO()
+    stderr_buf = StringIO()
+
+    old_stdout = sys.stdout
+    old_stderr = sys.stderr
+    sys.stdout = stdout_buf
+    sys.stderr = stderr_buf
+
+    try:
+        try:
+            code_obj = compile(cmd, "<remote>", "eval")
+            mode = "eval"
+            log("Compiled as eval.")
+        except SyntaxError:
+            code_obj = compile(cmd, "<remote>", "exec")
+            mode = "exec"
+            log("Compiled as exec.")
+
+        try:
+            if mode == "eval":
+                result = eval(code_obj, ctx, ctx)
+                if result is not None:
+                    print(_pp.pformat(result))
+            else:
+                exec code_obj in ctx, ctx
+        except SystemExit:
+            raise
+        except Exception:
+            log("Exception while running code.")
+            traceback.print_exc(file=stderr_buf)
+    finally:
+        sys.stdout = old_stdout
+        sys.stderr = old_stderr
+
+    out = stdout_buf.getvalue()
+    err = stderr_buf.getvalue()
+
+    if out:
+        log("stdout=%r" % out)
+    if err:
+        log("stderr=%r" % err)
+
+    return out, err
+
+
+# -----------------------------------------------------
+#  ACCEPT CONNECTION (reuse Stage-1 listener socket!)
+# -----------------------------------------------------
+buf1 = alloc(1)
+
+client_sock = u64_to_i64(sc.syscalls.accept(s, sockaddr_in, len_buf))
```

Comment:
maybe put the whole socket/while loop logic into another `while True` so user can reconnect to shell anytime